src/Product/ProductInfo.cs

```diff
@@ -8,44 +8,42 @@ namespace Azure.DataApiBuilder.Product;
 
 public static class ProductInfo
 {
-    public const string DEFAULT_VERSION = "0.0.0";
     public const string DAB_APP_NAME_ENV = "DAB_APP_NAME_ENV";
-    public static readonly string DEFAULT_APP_NAME = $"dab_oss_{ProductInfo.GetProductVersion()}";
-    public static readonly string ROLE_NAME = "DataApiBuilder";
+    public static readonly string DAB_USER_AGENT = $"dab_oss_{GetProductVersion()}";
+    public static readonly string CLOUD_ROLE_NAME = "DataApiBuilder";
 
     /// <summary>
-    /// Reads the product version from the executing assembly's file version information.
+    /// Returns the Product version in Major.Minor.Patch format without a commit hash.
+    /// FileVersionInfo.ProductBuildPart is used to represent the Patch version.
+    /// FileVersionInfo is used to retrieve the version information from the executing assembly
+    /// set by the Version property in Directory.Build.props
     /// </summary>
-    /// <returns>Product version if not null, default version 0.0.0 otherwise.</returns>
-    public static string GetProductVersion()
+    /// <param name="includeCommitHash">If true, returns the version string with the commit hash</param>
+    /// <returns>Version string "Major.Minor.Patch"</returns>
+    public static string GetProductVersion(bool includeCommitHash = false)
     {
         Assembly assembly = Assembly.GetExecutingAssembly();
         FileVersionInfo fileVersionInfo = FileVersionInfo.GetVersionInfo(assembly.Location);
-        string? version = fileVersionInfo.ProductVersion;
 
-        return version ?? DEFAULT_VERSION;
+        if (includeCommitHash)
+        {
+            return fileVersionInfo.ProductVersion ?? "DAB_UNVERSIONED";
```

Comment:
will there be any scenario where it will return DAB_UNVERSIONED? Also why did we choose this over "0.0.0"?

Reply:
Agree unversioned is not helpful. I'll come up with better value. I added that value because `fileVersionInfo.ProductVersion` is nullable.

 Not using `0.0.0` because that does not accurately reflect the version.